### PR TITLE
*: remove TiDB Binlog after deprecation

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

Starting from TiDB v8.4.0, TiDB Binlog is removed after deprecation. This PR removes and updates documents related to removing TiDB Binlog.

**Note:** 
- The TiDB Binlog line in [TiDB Features | TiDB Docs](https://docs.pingcap.com/tidb/dev/basic-features) will be updated in a separate PR that adds a new v8.4 column to `basic-features.md` and updates the status of all affected features. This PR doesn't update it.
- I've added `aliases` for 21 deleted `.md` files in PR https://github.com/pingcap/docs-cn/pull/18625 for docs in Chinese, to avoid broken 404 links. All the URLs of deleted files will be redirected to those links of v8.3. And will create such a separate PR for deleted docs in English.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18576
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -519,7 +519,7 @@ Configuration items related to performance.
 
 - The size limit of a single transaction in TiDB.
 - Default value: `104857600` (in bytes)
-- In a single transaction, the total size of key-value records cannot exceed this value. The maximum value of this parameter is `1099511627776` (1 TB). Note that if you have used the binlog to serve the downstream consumer Kafka (such as the `arbiter` cluster), the value of this parameter must be no more than `1073741824` (1 GB). This is because 1 GB is the upper limit of a single message size that Kafka can process. Otherwise, an error is returned if this limit is exceeded.
+- In a single transaction, the total size of key-value records cannot exceed this value. The maximum value of this parameter is `1099511627776` (1 TB). Note that if you have used the binlog to serve the downstream consumer Kafka, the value of this parameter must be no more than `1073741824` (1 GB). This is because 1 GB is the upper limit of a single message size that Kafka can process. Otherwise, an error is returned if this limit is exceeded.
```
[the remark this answers is not in the document]
Comment (lilin90) [reply tidb-configuration-file.md:522]: This change is aligned with https://github.com/pingcap/docs-cn/pull/18576/commits/bba75f618a7874935861ced69d28f7bb96934d0a.

**Comment by lilin90** [comment]
@Benjamin2037 has confirmed the new commits by @hfxsd that delete more descriptions about Pump and Drainer. The updates are the same as those for Chinese docs (see https://github.com/pingcap/docs-cn/pull/18576#issuecomment-2357611610).

**Review by hfxsd — APPROVED** [review]

**Pull request merged**